app/models/operatingsystem.rb

```diff
@@ -313,6 +313,28 @@ def pxe_kernel_options(params)
     options
   end
 
+  def last_modified_id(media_provider)
```

Comment:
The same operating system can be installed from multiple media sources (even before we start the "flavours" saga). I think this shouldn't be on operating system object. Can I suggest moving this method to medium provider's `unique_id` and caching the whole medium provider? As for Katello, from talking to them, it seems that `kickstart_repo.id` is unique and will change with each promotion.

Reply:
I will move it but that will create a problem of making HTTP HEAD call several times per request. I found a way how to memorize media provider by moving `pxe_files` from OS to host and with a slight change in renderer.

As for Katello, promotion is irrelevant and we need to do what I suggest anyway. When you sync new RHEL into Library, there is no promotion involved so the ID stays the same, yet there is a new PXE files pair which will likely break `wget`.